overlord/snapstate/snapstate.go

```diff
@@ -1297,7 +1305,11 @@ func InstallMany(st *state.State, names []string, revOpts []*RevisionOptions, us
 
 	var transactionLane int
 	if flags.Transaction == client.TransactionAllSnaps {
-		transactionLane = st.NewLane()
+		if flags.Lane != 0 {
+			transactionLane = flags.Lane
```

Comment:
should it be an error to have set if  flags.Transaction != client.TransactionAllSnaps ?

Reply:
Sure, I can change it to error if we only want to allow the `Lane` with transaction or I can always set the `Lane` even if there's no transaction. That also doesn't seem conceptually incorrect, we just don't have a use case for it now